Purchases/Public/Purchases.swift

```diff
@@ -617,6 +617,31 @@ public extension Purchases {
         }
     }
 
+    /**
+     * This function will logIn the current user with an appUserID.
+     *
+     * - Parameter appUserID: The appUserID that should be linked to the current user.
+     *
+     * The callback will be called with the latest CustomerInfo for the user, as well as a boolean
+     * indicating whether the user was created for the first time in the RevenueCat backend.
+     * See https://docs.revenuecat.com/docs/user-ids
+     */
+    @available(iOS 15.0, macOS 12, tvOS 15.0, watchOS 8.0, *)
+    func logIn(_ appUserID: String) async throws -> (CustomerInfo, Bool) {
```

Comment:
Part of me feels like this should be in a separate file like `Purchases+async.swift` but also, not really because we should probably keep `logIn` funcs together. So I actually think I want to argue for placing this `logIn` function next to the original one. 
Or- we could rename this func to something else, make it private, and then have a wrapper public `func logIn(_ appUserID: String) async throws -> (CustomerInfo, Bool)` func that calls it. We place that wrapper func next to the original `logIn` func so that the two `logIn` funcs can be next to eachother without having the entire function body body there, too.

Reply:
yeah, I think that's a good idea. That way the boilerplate of the checkedContinuations is kept in a single file and this one gets less noise. 
And I think we might wanna consider doing that for other stuff too eventually. This file is really big and it's hard to read the public API from it.

Reply:
I'm going to do an initial pass adding them all, and then a second pass cleaning this up and separating them into a separate file with a private implementation

Reply:
Sounds great!